components/service/glean/src/test/java/mozilla/components/service/glean/storages/DatetimesStorageEngineTest.kt

```diff
@@ -0,0 +1,167 @@
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.eq
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+import org.robolectric.RobolectricTestRunner
+import java.util.Calendar
+import java.util.TimeZone
+
+@RunWith(RobolectricTestRunner::class)
+class DatetimesStorageEngineTest {
+
+    @Before
+    fun setUp() {
+        DatetimesStorageEngine.applicationContext = ApplicationProvider.getApplicationContext()
+        // Clear the stored "user" preferences between tests.
+        ApplicationProvider.getApplicationContext<Context>()
+            .getSharedPreferences(DatetimesStorageEngine.javaClass.simpleName, Context.MODE_PRIVATE)
+            .edit()
+            .clear()
+            .apply()
+        DatetimesStorageEngine.clearAllStores()
+    }
+
+    @Test
+    fun `datetime deserializer should correctly parse datetimes`() {
+        val persistedSample = mapOf(
+            "store1#telemetry.invalid_date" to "2019-13-01T00:00:00Z",
+            "store1#telemetry.valid_date1" to "1993-02-23T09:05:23-08:00",
+            "store1#telemetry.null" to null,
+            "store1#telemetry.invalid_string" to "test"
+        )
+
+        val storageEngine = DatetimesStorageEngineImplementation()
+
+        // Create a fake application context that will be used to load our data.
+        val context = mock(Context::class.java)
+        val sharedPreferences = mock(SharedPreferences::class.java)
+        `when`(sharedPreferences.all).thenAnswer { persistedSample }
+        `when`(context.getSharedPreferences(
+            eq(storageEngine::class.java.simpleName),
+            eq(Context.MODE_PRIVATE)
+        )).thenReturn(sharedPreferences)
+
+        storageEngine.applicationContext = context
+        val snapshot = storageEngine.getSnapshot(storeName = "store1", clearStore = true)
+        assertEquals(1, snapshot!!.size)
+        assertEquals("1993-02-23T09:05:23-08:00", snapshot["telemetry.valid_date1"])
+    }
+
+    @Test
+    fun `datetime serializer should correctly serialize datetimes`() {
+        val value = Calendar.getInstance()!!
+        value.set(1993, 1, 23, 9, 5, 23)
+        value.set(Calendar.MILLISECOND, 0)
+        value.setTimeZone(TimeZone.getTimeZone("America/Los_Angeles"))
+
+        run {
+            val storageEngine = DatetimesStorageEngineImplementation()
+            storageEngine.applicationContext = ApplicationProvider.getApplicationContext()
+
+            val metric = DatetimeMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.User,
+                name = "datetime_metric",
+                sendInPings = listOf("store1")
+            )
+
+            // Record the datetime in the store, without providing optional arguments.
+            storageEngine.set(metric, value)
+
+            // Get the snapshot from "store1"
+            val snapshot = storageEngine.getSnapshotAsJSON(storeName = "store1",
```

Comment:
Since you created `metric` above as a `DatetimeMetricType` can you use the test API functions here?